### PR TITLE
refactor: change pgn mate score format from moves to plies

### PR DESCRIPTION
**Pull request by gahtan-syarif**
i think this is the correct conversion?
based on SF's `auto m = (mate.plies > 0 ? (mate.plies + 1) : mate.plies) / 2;`

### CODE REVIEW

**Comment by gahtan-syarif** [comment]
ok, verified it with cutechess and its correct

File: app/src/matchmaking/match/match.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ void Match::addMoveData(const Player& player, int64_t measured_time_ms, bool leg
         ss << (move_data.score >= 0 ? '+' : '-');
         ss << std::fixed << std::setprecision(2) << (float(std::abs(move_data.score)) / 100);
     } else if (score_type == engine::ScoreType::MATE) {
-        uint64_t plies = board_.sideToMove() == Chess::Color::WHITE ? std::abs(move_data.score) * 2 
+        uint64_t plies = board_.sideToMove() == chess::Color::WHITE ? std::abs(move_data.score) * 2
```
Comment (vondele) [inline app/src/matchmaking/match/match.cpp]: `chess::Color::WHITE` doesn't this depend on which side is winning ?

Comment (gahtan-syarif) [reply app/src/matchmaking/match/match.cpp]: yeah, this one is outdated, this is the new one:
```
uint64_t plies = move_data.score > 0 ? std::abs(move_data.score) * 2 - 1
                                             : std::abs(move_data.score) * 2;
```

File: app/src/matchmaking/match/match.cpp [hunk 2]
```diff
@@ -52,7 +52,9 @@ void Match::addMoveData(const Player& player, int64_t measured_time_ms, bool leg
         ss << (move_data.score >= 0 ? '+' : '-');
         ss << std::fixed << std::setprecision(2) << (float(std::abs(move_data.score)) / 100);
     } else if (score_type == engine::ScoreType::MATE) {
-        ss << (move_data.score > 0 ? "+M" : "-M") << std::to_string(std::abs(move_data.score));
+        uint64_t plies = move_data.score > 0 ? std::abs(move_data.score) * 2 - 1
```
Comment (robertnurnberg) [inline app/src/matchmaking/match/match.cpp:55]: Abs not needed here I think. Similarly below.

Comment (gahtan-syarif) [reply app/src/matchmaking/match/match.cpp]: Right, its always positive, fixed 👍

**Pull request merged**